pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -72,12 +72,12 @@ func (u UsernamePassword) Setup(baseDir string) error {
 
```

Comment:
why do we continue if we had no username or password? We could have hit step 3 above and defaulted to `DefaultUsername`, and gotten no password (from either the secret or the URL), and we still continue and write a credential gitconfig. We should only do that if we have password (from the URL or from the secret), or if we have a username (from the secret), right?